### ISSUE TITLE
Electron example fails with error TypeError: Cannot read property 'filter' of undefined

### ISSUE THREAD

**Issue by Patronics** [mozilla/STT-examples#71]
Running the electron example as described in it's readme (the only deviation from the instructions was to run `npm audit fix` to update packages with security risks, as advised by npm), I get an electron dialog 

```
A JavaScript error occurred in the main process

Uncaught Exception:
TypeError: Cannot read property 'filter' of undefined
    at /Users/me/path/to/DeepSpeech-examples/electron/public/create-window.js:72:20
    at FSReqCallback.oncomplete (fs.js:146:23)

```

the output on the comandline itself is 
```
Debugger listening on ws://127.0.0.1:5858/6859e332-415b-4033-a926-57b854cd4237
[1] For help, see: https://nodejs.org/en/docs/inspector
[1] TensorFlow: v2.2.0-17-g0854bb5188
[1] DeepSpeech: v0.8.0-0-gf56b07da
[1] 2020-08-08 19:01:19.581495: I tensorflow/core/platform/cpu_feature_guard.cc:143] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
[1] model loaded
[0] 
[0] Could not open App.js in the editor.
[0] 
[0] To set up the editor integration, add something like REACT_EDITOR=atom to the .env.local file in your project folder and restart the development server. Learn more: https://goo.gl/MMTaZt
```

Additionally, if I attempt to connect to localhost:3000 as the program suggests, I get a different error page, saying 

```
TypeError: undefined is not an object (evaluating 'window.ipcRenderer.invoke')

componentDidMount
src/App.js:16
  13 | 
  14 | componentDidMount() {
  15 | 	// when the component mounts, get the list of .wav files
> 16 | 	window.ipcRenderer.invoke('load-files')
     | ^  17 | 	.then(files => {
  18 | 		console.log('files', files);
  19 | 		this.setState({

▼ 15 stack frames were expanded.
commitLifeCycles
node_modules/react-dom/cjs/react-dom.development.js:19814
commitLayoutEffects
node_modules/react-dom/cjs/react-dom.development.js:22803
callCallback
node_modules/react-dom/cjs/react-dom.development.js:188
invokeGuardedCallbackDev
node_modules/react-dom/cjs/react-dom.development.js:237
invokeGuardedCallback
node_modules/react-dom/cjs/react-dom.development.js:292
commitRootImpl
node_modules/react-dom/cjs/react-dom.development.js:22541
unstable_runWithPriority
node_modules/scheduler/cjs/scheduler.development.js:653
commitRoot
node_modules/react-dom/cjs/react-dom.development.js:22381
finishSyncRender
node_modules/react-dom/cjs/react-dom.development.js:21807
performSyncWorkOnRoot
node_modules/react-dom/cjs/react-dom.development.js:21793
scheduleUpdateOnFiber
node_modules/react-dom/cjs/react-dom.development.js:21188
updateContainer
node_modules/react-dom/cjs/react-dom.development.js:24373
(anonymous function)
node_modules/react-dom/cjs/react-dom.development.js:24758
unbatchedUpdates
node_modules/react-dom/cjs/react-dom.development.js:21903
legacyRenderSubtreeIntoContainer
node_modules/react-dom/cjs/react-dom.development.js:24757


./src/index.js
src/index.js:5
  2 | import ReactDOM from 'react-dom';
  3 | import App from './App';
  4 | 
> 5 | ReactDOM.render(<App/>, document.getElementById('root'));

__webpack_require__
/Users/me/path/to/DeepSpeech-examples/electron/webpack/bootstrap:784
  781 | };
  782 | 
  783 | // Execute the module function
> 784 | modules[moduleId].call(module.exports, module, module.exports, hotCreateRequire(moduleId));
      | ^  785 | 
  786 | // Flag the module as loaded
  787 | module.l = true;

fn
/Users/me/path/to/DeepSpeech-examples/electron/webpack/bootstrap:150
  147 | 		);
  148 | 		hotCurrentParents = [];
  149 | 	}
> 150 | 	return __webpack_require__(request);
      | ^  151 | };
  152 | var ObjectFactory = function ObjectFactory(name) {
  153 | 	return {
View compiled
(anonymous function)
http://localhost:3000/static/js/main.chunk.js:134:37
__webpack_require__
/Users/me/path/to/DeepSpeech-examples/electron/webpack/bootstrap:784
  781 | };
  782 | 
  783 | // Execute the module function
> 784 | modules[moduleId].call(module.exports, module, module.exports, hotCreateRequire(moduleId));
      | ^  785 | 
  786 | // Flag the module as loaded
  787 | module.l = true;
View compiled
checkDeferredModules
/Users//me/path/to/DeepSpeech-examples/electron/webpack/bootstrap:45
  42 | 	}
  43 | 	if(fulfilled) {
  44 | 		deferredModules.splice(i--, 1);
> 45 | 		result = __webpack_require__(__webpack_require__.s = deferredModule[0]);
     | ^  46 | 	}
  47 | }
  48 | 
View compiled
webpackJsonpCallback
/Users/me/path/to/DeepSpeech-examples/electron/webpack/bootstrap:32
  29 | 	deferredModules.push.apply(deferredModules, executeModules || []);
  30 | 
  31 | 	// run deferred modules when all chunks ready
> 32 | 	return checkDeferredModules();
     | ^  33 | };
  34 | function checkDeferredModules() {
  35 | 	var result;
View compiled
global code
http://localhost:3000/static/js/main.chunk.js:1:95
```

How  can I fix this?

**Comment by lissyx** [comment]
@dsteinman do you have an idea ?

**Comment by dsteinman** [comment]
```
files = files.filter(function (file) {
```

@Patronics I'm pretty sure you just forgot to download the audio.tar.gz file.  It needs to be extracted to the `public/audio` directory.

I guess maybe I should add a proper warning there.

**Comment by lissyx** [comment]
I guess this is fixed per #72 , but @dsteinman can you ensure it is also on master?

**Comment by dsteinman** [comment]
Yes it's there:

https://github.com/mozilla/STT-examples/blob/r0.8/electron/public/create-window.js#L81

**Comment by lissyx** [comment]
> Yes it's there:
> 
> https://github.com/mozilla/STT-examples/blob/r0.8/electron/public/create-window.js#L81

That's r0.8, do you mind verifying on master as well?

**Comment by dsteinman** [comment]
Oh, yeah looks like it was only merged to 0.8, it's not there:

https://github.com/mozilla/STT-examples/blob/master/electron/public/create-window.js
